File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <string>
using namespace std;

unordered_map<string, int> um;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    for(auto &c: completion){
        um[c]++;
    }
    
    for(auto p: participant){
        auto it = um.find(p);
        if(it != um.end()){
            if(it->second == 1)
                um.erase(it->first);
            else
                it->second--;
        }
        else {
            answer = p;
        }
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const hash = {}
    
    for(let i = 0; i < completion.length; i++) {
        if(!hash[completion[i]]) hash[completion[i]] = 1;
        else hash[completion[i]]++;
    }
    
    for(let i = 0; i < participant.length; i++) {
        if(!hash[participant[i]]) return participant[i];
        if(hash[participant[i]]) hash[participant[i]]--;
    }
}